src/ui/control/fullscreen_control.js

```diff
@@ -7,26 +7,36 @@ import window from '../../util/window';
 
 import type Map from '../map';
 
+type Options = {
+    source?: string
+};
+
 /**
  * A `FullscreenControl` control contains a button for toggling the map in and out of fullscreen mode.
  *
  * @implements {IControl}
+ * @param {Object} [options]
+ * @param {string} [options.source] `source` is a string representing the DOM element which should be made full screen. It must be a [full screen compatible HTML element](https://developer.mozilla.org/en-US/docs/Web/API/Element/requestFullScreen#Compatible_elements). By default, the map container element will be made full screen.
+ *
  * @example
- * map.addControl(new mapboxgl.FullscreenControl());
+ * map.addControl(new mapboxgl.FullscreenControl({source: 'body'}));
  * @see [View a fullscreen map](https://www.mapbox.com/mapbox-gl-js/example/fullscreen/)
  */
 
 class FullscreenControl {
     _map: Map;
-    _mapContainer: HTMLElement;
     _container: HTMLElement;
     _fullscreen: boolean;
     _fullscreenchange: string;
     _fullscreenButton: HTMLElement;
     _className: string;
+    _source: HTMLElement;
 
-    constructor() {
+    constructor(options: Options) {
         this._fullscreen = false;
+        if (options && options.source) {
+            this._source = window.document.querySelector(options.source);
```

Comment:
Our other APIs that need an HTML element reference, such as Map `container` and Marker `element`, accept an `HTMLElement` rather than a selector — I'd suggest keeping this consistent.